predicators/datasets/vlm_input_data.py

```diff
@@ -0,0 +1,243 @@
+                            if num_args == 0:
+                                num_args = len(obj_args)
+                            else:
+                                assert num_args == len(obj_args)
+                        # Given this, add one new predicate with num_args
+                        # number of 'object' type arguments.
+                        pred_name_to_pred[pred_name] = Predicate(
+                            pred_name, [obj_type for _ in range(num_args)],
+                            _stripped_classifier)
+
+                # Given that we've now built up predicates and object
+                # dictionaries. We can now convert the current state into
+                # ground atoms!
+                for obj_args, truth_value in objs_and_val_dict.items():
+                    if truth_value:
+                        curr_ground_atoms_state.add(
+                            GroundAtom(
+                                pred_name_to_pred[pred_name],
+                                [curr_obj_name_to_obj[o] for o in obj_args]))
+            curr_atoms_traj.append(curr_ground_atoms_state)
+        # Add the goal atom at the end of the trajectory.
+        curr_atoms_traj[-1].add(goal_atom)
+        atoms_trajs.append(curr_atoms_traj)
+    return atoms_trajs
+
+
+def _parse_structured_actions_into_ground_options(
+        structuredactions_trajs: List[List[Tuple[str, Tuple[str, ...],
+                                                 List[float]]]],
+        known_options: Set[ParameterizedOption],
+        train_tasks: List[Task]) -> List[List[_Option]]:
+    """Convert structured actions trajectories into actual lists of ground
+    options trajectories."""
+    assert len(structuredactions_trajs) == len(train_tasks)
+    option_name_to_option = {o.name: o for o in known_options}
+    option_trajs = []
+    for i, traj in enumerate(structuredactions_trajs):
+        curr_obj_name_to_obj = {
+            obj.name: obj
+            for obj in set(train_tasks[i].init)
+        }
+        curr_option_traj = []
+        for structured_action in traj:
+            option = option_name_to_option[structured_action[0]]
+            ground_option = option.ground([
+                curr_obj_name_to_obj[obj_name]
+                for obj_name in structured_action[1]
+            ], np.array(structured_action[2]))
+            # Call initiable here because we will be calling
+            # terminal later, and initiable always needs
+            # to be called first.
+            ground_option.initiable(train_tasks[i].init)
+            curr_option_traj.append(ground_option)
+        option_trajs.append(curr_option_traj)
+    return option_trajs
+
+
+def _create_dummy_goal_state_for_each_task(
+        env: BaseEnv, train_tasks: List[Task]) -> List[State]:
+    """Uses a lot of assumptions to generate a state in which a dummy goal
+    predicate holds for each train task."""
+    # We assume there is only one goal predicate, and that it is
```

Comment:
is this going to be a long-term assumption?

Reply:
no - just for now!